renderer/reducers/utils/waitForIpc.js

```diff
@@ -0,0 +1,52 @@
+import { send } from 'redux-electron-ipc'
+import promiseTimeout from '@zap/utils/promiseTimeout'
+
+/**
+ * waitForIpcEvent - Sends a message to main process and waits for the reply.
+ *
+ * @param {string} message ipc message name
+ * @param {object} params ipc event params
+ * @param {string} responseEvent ipc success response event. defaults to `message`
+ * @param {string} failureEvent ipc failure response event. defaults to `${event}Failure`
+ * @param {number} timeout timeout for the ipc response
+ * @returns {Promise} promise that fulfills on success/failure response from ipc main or
+ * timeouts after `timeout` milliseconds
+ */
+export default function waitForIpcEvent(
+  message,
+  params,
+  responseEvent,
+  failureEvent,
+  timeout = 5000
+) {
+  return dispatch => {
+    const failureEventName = failureEvent || `${message}Failure`
+    const successEventName = responseEvent || message
+
+    const removeFailureEventListeners = () =>
+      window.ipcRenderer.removeAllListeners(failureEventName)
```

Comment:
Calling `removeAllListeners` could have unintended consequences, removing listeners unrelated to this particular call to `waitForIpcEvent`.

We should take care to only remove the specific listeners that were added by us.